<launch>

    <!-- This transforms the odom topic of the realsense to the base link frame -->
    <node pkg="localization" type="rs_to_odom.py" name="rs_to_odom"  output="screen">
        <param name="t265_odom_frame" value="/t265_odom_frame"/>
        <param name="fake_base_link" value="/fake_base_link"/>
        <param name="input_odom" value="/t265/odom/sample"/>
        <param name="output_odom" value="/rs_odom"/>
    </node>
    
    <remap from="odom" to="/rs_odom_reset"/>

    <node pkg="robot_pose_ekf" type="robot_pose_ekf" name="robot_pose_ekf">
        <param name="output_frame" value="odom"/>
        <param name="freq" value="30.0"/>
        <param name="sensor_timeout" value="1.0"/>
        <param name="odom_used" value="true"/>
        <param name="imu_used" value="false"/>
        <param name="vo_used" value="false"/>
        <param name="debug" value="false"/>
        <param name="self_diagnose" value="false"/>
    </node>

    <node pkg="localization" type="set_pose.py" name="set_pose"  output="screen">
        <rosparam command="load" file="$(find localization)/config/rs_set_pose.yaml"/>
    </node>
    
</launch>